WalletWasabi.Fluent.Generators/NavigationMetaDataGenerator.cs

```diff
@@ -90,7 +90,7 @@ public void Execute(GeneratorExecutionContext context)
 		{
 			context.AddSource("NavigationMetaDataAttribute", SourceText.From(AttributeText, Encoding.UTF8));
 
-			if (!(context.SyntaxReceiver is SyntaxReceiver receiver))
+			if (context.SyntaxReceiver is not SyntaxReceiver receiver)
```

Comment:
I don't trust on this pattern, it is not clear how it works. Are you 100% sure that this `receiver` variable is finally assigned? Have to tested it?

Reply:
I am not sure how I can properly test this, I get `the breakpoint will not currently be hit no symbols have been loaded for this document` when trying to debug it.

I found this: https://devblogs.microsoft.com/dotnet/c-9-0-on-the-record/#logical-patterns
> And in fact, in an is not expression like that we allow you to name the Customer for subsequent use:
> 
> ```
> if (e is not Customer c) { throw ... } // if this branch throws or returns...
> var n = c.FirstName; // ... c is definitely assigned here
> ```

Reply:
> I am not sure how I can properly test this, I get `the breakpoint will not currently be hit no symbols have been loaded for this document` when trying to debug it.

Uncomment this line and build, this should attach debugger:
https://github.com/zkSNACKs/WalletWasabi/blob/1e40eb5dd926dd80bfd69e31e657435925e1b006/WalletWasabi.Fluent.Generators/NavigationMetaDataGenerator.cs#L85

Reply:
Can you write a test code here @yahiheb? 

https://dotnetfiddle.net/

Tip: Instead of `SyntaxReceiver` use `StringBuilder` class in the code example.

Reply:
I believe @yahiheb's changes are correct.

For me a proof of correctness is rather to compare IL instructions generated by C# compiler:

* [original pattern](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgAwFgBQaCMOy6ABMngHQAqApgB7ADcOOA9gEYBW1AxsMQM7EAvMQBEACTABCUU2yE8ATgAUomvz7cAhv2rEKogJRycYAGbFlU5YLCCAysABOYAHYBzAEIBXSDGpOpHjoPn4BeIaGOADeOMTxQSqiAJIA5AC2xK4sfI4uHqEQ/k7kRnIAvjhAA===)
* [proposed pattern](https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgAwFgBQaCMOy6ABMngHQAqApgB7ADcOOA9gEYBW1AxsMQM7EAvMQBEACTABCUU2yE8ATgAUomvz7cAhv2rFU5UQEo5OMADNiywWEEA7FnwDKwAE5g7AcwBCAV0gw1K6keOh+AUGoRjgA3jjECSEqogCSAOQAtsQOzm4ePv4Qga6GJjgAvkA=)